README.md

```diff
@@ -1,7 +1,29 @@
-druntime for Unikraft
-=============================
+# D runtime for Unikraft
 
-This is the port of druntime for Unikraft as external library.
+This is the port of the D runtime for Unikraft as external library.    
+
+This library has a number of dependencies, so please make sure you have the following libraries
+ added to the `LIBS` variable in your application's Makefile:
+* lib-compiler-rt
+* lib-libcxx
+* lib-libcxxabi
+* lib-libucontext
+* lib-libunwind
+* lib-gcc
+* lib-pthread-embedded
+* lib-newlib
+* lib-lwip
+* lib-druntime
+
+### Applications
+Applications may be run only in Linux user space. All mandatory dependencies are selected by default
+in the configuration menu. For building an application you may choose out of two compilers:
+* GDC -> use command `make D_COMPILER=gdc` for building. We provide a *Hello, World!* sample application in the `main.d` source file.
+* DMD -betterC -> use command `make D_COMPILER=dmd` for building. We provide a *Hello, World!* sample application that is betterC-compatible in the `betterC-main.d` source file.
+
+### Limitations
+1. There is no support for running D application images on top of KVM or XEN.
+2. There is no support for TLS. This has mainly two implications: multi-threaded applications will not properly function; global variables need to be declared by using the `__gshared` keyword, as this forces them to be placed in the bss/data sections.
```

Comment:
I would add here the fact that there is no support for compilation yet, and it needs to be added first.

Reply:
Done